x-pack/plugins/infra/public/alerting/metric_threshold/lib/generate_unique_key.ts

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { MetricExpression } from '../types';
+
+export const generateUniqueKey = (criterion: MetricExpression) => {
+  return criterion.aggType + criterion.comparator + criterion.threshold.join(',');
```

Comment:
Not sure if this is unique enough for all metrics 🤔 

cc @simianhacker

Reply:
Per our conversation, I would probably include `criterion.metric` when available to disambiguate multiple expressions with the same agg, comparator, and thresholds (which is unlikely but possible).